packages/client/wallets/aa/src/blockchain/wallets/smartAccount.ts

```diff
@@ -0,0 +1,49 @@
+import { logInfo } from "@/services/logging";
+import { gelatoBundlerProperties, usesGelatoBundler } from "@/utils";
+import { SmartAccountClient } from "permissionless";
+import { EntryPoint } from "permissionless/types/entrypoint";
+
+import { EVMBlockchainIncludingTestnet } from "@crossmint/common-sdk-base";
+
+export function toCrossmintSmartAccountClient<Client extends SmartAccountClient<EntryPoint>>({
+    crossmintChain,
+    smartAccountClient,
+}: {
+    crossmintChain: EVMBlockchainIncludingTestnet;
+    smartAccountClient: Client;
+}): Client {
+    return {
```

Comment:
I'm not a huge fan of the style here, w/ the inline functions defined this way. But looking through `permissionless.js` and the ZD SDK this is the canon way to do this.

Reply:
Out of curiosity, what other approach were you thinking of?

Reply:
Something more similar to the `LoggerWrapper` we use, otherwise if `viem` and `permissionsless` we're more OOO style using decorators or something like that.